src/freenet/support/Waiter.java

```diff
@@ -0,0 +1,23 @@
+package freenet.support;
+
+/** Waits for some other thread to signal to proceed. */
+public class Waiter {
```

Comment:
Why are you not using e.g. a `Semaphore` or a `CountDownLatch` here? Please don’t add custom implementations of classes that are already in the JDK.